### PR TITLE
A couple of helpers for streams that I've found helpful

### PR DESCRIPTION
**Pull request by daurnimator**

### CODE REVIEW

File: lib/std/io/in_stream.zig [hunk 1]
```diff
@@ -231,6 +234,18 @@ pub fn InStream(
             }
         }
 
+        /// Reads `slice.len` bytes from the stream and returns if they are the same as the passed slice
+        pub fn isBytes(self: Self, slice: []const u8) !bool {
```
Comment (data-man) [inline lib/std/io/in_stream.zig:238]: Rename to readWithCheck or skipWithCheck?

Comment (joachimschmidt557) [reply lib/std/io/in_stream.zig:238]: Or maybe `readEqual` or `readEql`?

Comment (data-man) [reply lib/std/io/in_stream.zig:238]: The first impression:
Q: stream.isBytes?
A: Sure, any stream is bytes.

**Comment by data-man** [comment]
isBytes is the very bad name.

**Comment by andrewrk** [comment]
Reverting

**Pull request merged**